Trying to compile with GPIO slowdown

trying to compile with GPIO slowdown on node 0.10.3 using node-gyp@3.2.1, but I'm running into this compile error...

```
Roys-iMac:node-rpi-rgb-led-matrix roykolak$ npm i

> node-rpi-rgb-led-matrix@0.1.0 install /Users/roykolak/Projects/bigdots/node-rpi-rgb-led-matrix
> node-gyp rebuild

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
make: *** No rule to make target `Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/bdf-font.o', needed by `Release/rpi-rgb-led-matrix.a'.  Stop.
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/roykolak/.nvm/v0.10.3/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:786:12)
gyp ERR! System Darwin 15.6.0
gyp ERR! command "node" "/Users/roykolak/.nvm/v0.10.3/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/roykolak/Projects/bigdots/node-rpi-rgb-led-matrix
gyp ERR! node -v v0.10.3
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok

npm ERR! Darwin 15.6.0
npm ERR! argv "node" "/Users/roykolak/.nvm/v0.10.3/bin/npm" "i"
npm ERR! node v0.10.3
npm ERR! npm  v3.10.7
npm ERR! code ELIFECYCLE
npm ERR! node-rpi-rgb-led-matrix@0.1.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-rpi-rgb-led-matrix@0.1.0 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the node-rpi-rgb-led-matrix package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-rpi-rgb-led-matrix
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls node-rpi-rgb-led-matrix
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/roykolak/Projects/bigdots/node-rpi-rgb-led-matrix/npm-debug.log
```

duh.... had to pull the submodule! Running into this now.. 

```
Roys-iMac:node-rpi-rgb-led-matrix roykolak$ node-gyp build
gyp info it worked if it ends with ok
gyp info using node-gyp@3.2.1
gyp info using node@0.10.3 | darwin | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/bdf-font.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/framebuffer.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/gpio.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/graphics.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/led-matrix.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/thread.o
../external/matrix/lib/thread.cc:54:5: error: unknown type name 'cpu_set_t'
    cpu_set_t cpu_mask;
    ^
1 error generated.
make: *** [Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/thread.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/roykolak/.nvm/v0.10.3/lib/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:786:12)
gyp ERR! System Darwin 15.6.0
gyp ERR! command "node" "/Users/roykolak/.nvm/v0.10.3/bin/node-gyp" "build"
gyp ERR! cwd /Users/roykolak/Projects/bigdots/node-rpi-rgb-led-matrix
gyp ERR! node -v v0.10.3
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok
```

was trying to compile on osx, but I jumped over to my raspberry pi and got further, but still ran into some issues. Any ideas?


```
pi@raspberrypi:~/node-rpi-rgb-led-matrix $ node-gyp build
gyp info it worked if it ends with ok
gyp info using node-gyp@3.4.0
gyp info using node@0.10.3 | linux | arm
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/pi/node-rpi-rgb-led-matrix/build'
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/bdf-font.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/framebuffer.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/gpio.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/graphics.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/led-matrix.o
  CXX(target) Release/obj.target/rpi-rgb-led-matrix/external/matrix/lib/thread.o
  AR(target) Release/obj.target/rpi-rgb-led-matrix.a
  COPY Release/rpi-rgb-led-matrix.a
  CXX(target) Release/obj.target/node-rpi-rgb-led-matrix/src/base.o
In file included from ../include/ledmatrix.h:16:0,
                 from ../src/base.cc:11:
../include/image.h:19:19: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   const uint8_t R();
                   ^
../include/image.h:20:19: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   const uint8_t G();
                   ^
../include/image.h:21:19: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   const uint8_t B();
                   ^
  CXX(target) Release/obj.target/node-rpi-rgb-led-matrix/src/ledmatrix.o
In file included from ../include/ledmatrix.h:16:0,
                 from ../src/ledmatrix.cc:15:
../include/image.h:19:19: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   const uint8_t R();
                   ^
../include/image.h:20:19: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   const uint8_t G();
                   ^
../include/image.h:21:19: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   const uint8_t B();
                   ^
In file included from /home/pi/.node-gyp/0.10.3/src/node_buffer.h:28:0,
                 from ../node_modules/nan/nan.h:48,
                 from ../include/ledmatrix.h:14,
                 from ../src/ledmatrix.cc:15:
../src/ledmatrix.cc: In static member function ‘static void LedMatrix::SetImageBuffer(const Nan::FunctionCallbackInfo<v8::Value>&)’:
../src/ledmatrix.cc:217:14: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  assert(bufl == width*height*3);
              ^
../src/ledmatrix.cc: In static member function ‘static void LedMatrix::UV_Scroll(uv_work_t*)’:
../src/ledmatrix.cc:324:22: error: ‘usleep’ was not declared in this scope
      usleep(uv->speed);
                      ^
../src/ledmatrix.cc:333:21: error: ‘usleep’ was not declared in this scope
     usleep(uv->speed);
                     ^
../src/ledmatrix.cc:341:21: error: ‘usleep’ was not declared in this scope
     usleep(uv->speed);
                     ^
../src/ledmatrix.cc:348:22: error: ‘usleep’ was not declared in this scope
      usleep(uv->speed);
                      ^
../src/ledmatrix.cc:358:22: error: ‘usleep’ was not declared in this scope
      usleep(uv->speed);
                      ^
../src/ledmatrix.cc:366:21: error: ‘usleep’ was not declared in this scope
     usleep(uv->speed);
                     ^
../src/ledmatrix.cc:374:21: error: ‘usleep’ was not declared in this scope
     usleep(uv->speed);
                     ^
../src/ledmatrix.cc:381:22: error: ‘usleep’ was not declared in this scope
      usleep(uv->speed);
                      ^
node-rpi-rgb-led-matrix.target.mk:102: recipe for target 'Release/obj.target/node-rpi-rgb-led-matrix/src/ledmatrix.o' failed
make: *** [Release/obj.target/node-rpi-rgb-led-matrix/src/ledmatrix.o] Error 1
make: Leaving directory '/home/pi/node-rpi-rgb-led-matrix/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:786:12)
gyp ERR! System Linux 4.4.13-v7+
gyp ERR! command "node" "/usr/bin/node-gyp" "build"
gyp ERR! cwd /home/pi/node-rpi-rgb-led-matrix
gyp ERR! node -v v0.10.3
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
```

--- a/binding.gyp
+++ b/binding.gyp
@@ -4,7 +4,7 @@
 	        "target_name": "node-rpi-rgb-led-matrix",
 	        "sources": [ "src/base.cc", "src/ledmatrix.cc", "src/image.cc" ],
 	        "dependencies": [ "./binding.gyp:rpi-rgb-led-matrix" ],
-	        "include_dirs": [ "./external/matrix/include", "./include/", "<!(node -e \"require('nan')\")" ]
+	        "include_dirs": [ "./external/matrix/include", "./include/", "<!(nodejs -e \"require('nan')\")" ]
 	    },
 		{
 			"target_name": "rpi-rgb-led-matrix",

--- a/include/image.h
+++ b/include/image.h
@@ -16,9 +16,9 @@ class Pixel {
 		// Pixel() : red(0), green(0), blue(0) {}
 		Pixel();
 
-		const uint8_t R();
-		const uint8_t G();
-		const uint8_t B();
+		uint8_t R() const;
+		uint8_t G() const;
+		uint8_t B() const;
 
 		void SetR(uint8_t r);
 		void SetG(uint8_t g);

--- a/src/image.cc
+++ b/src/image.cc
@@ -14,9 +14,9 @@
 
 Pixel::Pixel() : red(0), green(0), blue(0) {}
 
-const uint8_t Pixel::R() { return red; }
-const uint8_t Pixel::G() { return green; }
-const uint8_t Pixel::B() { return blue; }
+uint8_t Pixel::R() const { return red; }
+uint8_t Pixel::G() const { return green; }
+uint8_t Pixel::B() const { return blue; }
 
 void Pixel::SetR(uint8_t r) { red = r; }
 void Pixel::SetG(uint8_t g) { green = g; }

--- a/src/ledmatrix.cc
+++ b/src/ledmatrix.cc
@@ -10,6 +10,7 @@
 #include <node.h>
 
 #include <stdlib.h>
+#include <unistd.h>
 
 #include <led-matrix.h>
 #include <ledmatrix.h>
@@ -161,7 +162,7 @@ void LedMatrix::GetHeight(const Nan::FunctionCallbackInfo<v8::Value>& args) {
 void LedMatrix::SetPixel(const Nan::FunctionCallbackInfo<v8::Value>& args) {
 	LedMatrix* matrix = ObjectWrap::Unwrap<LedMatrix>(args.Holder());
 
-	if(!args.Length() == 5 || !args[0]->IsNumber() || !args[1]->IsNumber() || !args[2]->IsNumber()
+	if(args.Length() != 5 || !args[0]->IsNumber() || !args[1]->IsNumber() || !args[2]->IsNumber()
 	|| !args[3]->IsNumber() || !args[4]->IsNumber()) {
 		Nan::ThrowTypeError("Wrong parameters! Expects 5 numbers");
   	}
@@ -192,7 +193,7 @@ void LedMatrix::Clear(const Nan::FunctionCallbackInfo<v8::Value>& args) {
 void LedMatrix::Fill(const Nan::FunctionCallbackInfo<v8::Value>& args) {
 	LedMatrix* matrix = ObjectWrap::Unwrap<LedMatrix>(args.Holder());
 
-	if(!args.Length() == 3 || !args[0]->IsNumber() || !args[1]->IsNumber() || !args[2]->IsNumber()) {
+	if(args.Length() != 3 || !args[0]->IsNumber() || !args[1]->IsNumber() || !args[2]->IsNumber()) {
 		Nan::ThrowTypeError("Wrong parameters! Expects 3 numbers");
 	}
 
@@ -214,7 +215,7 @@ void LedMatrix::SetImageBuffer(const Nan::FunctionCallbackInfo<v8::Value>& args)
 	int width = args[1]->ToInteger()->Value();
 	int height = args[2]->ToInteger()->Value();
 
-	assert(bufl == width*height*3);
+	assert((int)bufl == width*height*3);
 
 	Image* img = new Image();
 	Pixel* pixels = (Pixel*) malloc(sizeof(Pixel)*width*height);

